hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/JoinHashPhysicalRel.java

```diff
@@ -58,9 +58,29 @@ public Join copy(
         return new JoinHashPhysicalRel(getCluster(), traitSet, left, right, conditionExpr, joinType);
     }
 
+    /**
+     * Cost calculation of Hash Join relation. It does not rely on children cost.
+     * <p>
+     * Hash Join algorithm is more advanced join algorithm, where it builds a hash table for left row set, and then
+     * compare each row from the right side
+     * Speaking of cost estimation, we are accounting the next properties:
+     * - row count is estimating ans L + R, because we traverse both sides once per JOIN.
+     * - same for CPU cost estimation multiplied by cost to build a hash table and left and right rows comparison.
+     * <p>
+     * The perfect assumption also must include memory (what is important in case of hash table) and IO cost estimation,
+     * as well as a selectivity for a right row set.
+     */
     @Override
     @Nullable
     public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
-        return super.computeSelfCost(planner, mq).multiplyBy(COST_FACTOR);
+        double leftRowCount = mq.getRowCount(getLeft());
+        double rightRowCount = mq.getRowCount(getRight());
+        double rowCount = leftRowCount + rightRowCount;
```

Comment:
in NLJ you multiply

Reply:
Yes. Check out javadoc for this method, please.

Reply:
ok, it is fixed now